src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -161,28 +176,97 @@ public void completed(Integer resultLen, Void ignore) {
                                 deleteFile(file);
                             }
 
-                            callback.onPutComplete();
+                            callback.onComplete();
                         }
 
                         @Override
                         public void failed(Throwable throwable, Void ignore) {
                             VolleyLog.e(
                                     throwable, "Failed to read file %s", file.getAbsolutePath());
                             deleteFile(file);
-                            callback.onPutComplete();
+                            callback.onComplete();
                             closeChannel(afc, "failed read");
                         }
                     });
         } catch (IOException e) {
             if (closeChannel(channel, "IOException")) {
                 deleteFile(file);
             }
-            callback.onPutComplete();
+            callback.onComplete();
+        }
+    }
+
+    /** Clears the cache. Deletes all cached files from disk. */
+    @Override
+    public void clear(OnCompleteCallback callback) {
+        File[] files = mRootDirectorySupplier.get().listFiles();
+        if (files != null) {
+            for (File file : files) {
+                deleteFile(file);
+            }
+        }
+        mEntries.clear();
+        mTotalSize = 0;
+        VolleyLog.d("Cache cleared.");
+        callback.onComplete();
+    }
+
+    /** Initializes the cache. */
+    @Override
+    public void initialize(OnCompleteCallback callback) {
+        File rootDirectory = mRootDirectorySupplier.get();
+        if (!rootDirectory.exists()) {
+            if (!rootDirectory.mkdirs()) {
+                VolleyLog.e("Unable to create cache dir %s", rootDirectory.getAbsolutePath());
+            }
+            return;
+        }
+        File[] files = rootDirectory.listFiles();
```

Comment:
It's my understanding that as long as rootDirectory is a folder and not a file, the output of listFiles() will never be null. But it could be a good check just in case, as we do it in DiskBasedCache.

Reply:
@sphill99 According to the method docs, the check is needed:

`Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs.`

Reply:
oh thanks! just added it